test/test_code.py

```diff
@@ -338,15 +350,27 @@ def test_execute_code(use_docker=None):
     assert isinstance(image, str) or docker is None or os.path.exists("/.dockerenv") or use_docker is False
 
 
+@pytest.mark.skipif(docker_package_installed is False, reason="docker package not installed")
+def test_execute_code_with_custom_filename_on_docker():
+    exit_code, msg, image = execute_code("print('hello world')", filename="tmp/codetest.py", use_docker=True)
+    assert exit_code == 0 and msg == "hello world\n", msg
+    assert image == "python:tmp_codetest.py"
+
+
+@pytest.mark.skipif(docker_package_installed is False, reason="docker package not installed")
+def test_execute_code_with_misformed_filename_on_docker():
+    exit_code, msg, image = execute_code(
+        "print('hello world')", filename="tmp/codetest.py (some extra information)", use_docker=True
+    )
+    assert exit_code == 0 and msg == "hello world\n", msg
+    assert image == "python:tmp_codetest.py__some_extra_information_"
```

Comment:
Are these two tests skipped by the current build workflow?

Reply:
Yes. Though I tested locally, and they passed. We need a separate PR to address docker tests on Github Workflow.

Reply:
They are tested in the openai workflow. I guess we should install docker in one of the python versions in the ubuntu environments of the build workflow.

Reply:
Opened #1228 to add this.